wp-includes/class-wp-https-ui.php

```diff
@@ -0,0 +1,220 @@
+	/**
+	 * The expected value if the option is enabled.
+	 *
+	 * @var string
+	 */
+	const OPTION_SELECTED_VALUE = '1';
+
+	/**
+	 * The ID of the settings section.
+	 *
+	 * @var string
+	 */
+	const SETTING_ID = 'wp_upgrade_https';
+
+	/**
+	 * The HTTPS protocol
+	 *
+	 * @var string
+	 */
+	const HTTPS_PROTOCOL = 'https://';
+
+	/**
+	 * Inits the class.
+	 */
+	public function init() {
+		add_action( 'admin_init', array( $this, 'init_admin' ) );
+		add_action( 'admin_init', array( $this, 'filter_site_url_and_home' ) );
+	}
+
+	/**
+	 * Initializes the admin tasks.
+	 */
+	public function init_admin() {
+		$this->register_settings();
+		$this->add_settings_field();
+	}
+
+	/**
+	 * Registers the HTTPS settings.
+	 */
+	public function register_settings() {
+		$args = array(
+			'type'              => 'string',
+			'sanitize_callback' => array( $this, 'sanitize_callback' ),
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_HTTPS_OPTION,
+			$args
+		);
+
+		register_setting(
+			self::OPTION_GROUP,
+			self::UPGRADE_INSECURE_CONTENT_OPTION,
+			$args
+		);
+	}
+
+	/**
+	 * Sanitizes the raw setting value.
+	 *
+	 * @param string $raw_setting The setting before sanitizing it.
+	 * @return string|null The setting, or null if it's not valid.
+	 */
+	public function sanitize_callback( $raw_setting ) {
+		if ( self::OPTION_SELECTED_VALUE === $raw_setting || '0' === $raw_setting ) {
+			return $raw_setting;
+		}
+	}
+
+	/**
+	 * Adds a settings field to the 'Reading Settings' page.
+	 */
+	public function add_settings_field() {
+		add_settings_field(
+			self::SETTING_ID,
+			__( 'HTTPS', 'pwa' ),
+			array( $this, 'render_settings' ),
+			self::OPTION_GROUP
+		);
+	}
+
+	/**
+	 * Renders the HTTPS settings in /wp-admin on the Reading Settings page.
+	 */
+	public function render_settings() {
+		$upgrade_https_value           = get_option( self::UPGRADE_HTTPS_OPTION );
+		$upgrade_insecure_content      = get_option( self::UPGRADE_INSECURE_CONTENT_OPTION );
+		$https_more_details            = sprintf(
+			'<a href="%s">%s</a>',
+			'https://make.wordpress.org/support/user-manual/web-publishing/https-for-wordpress/',
+			esc_html__( 'More details', 'pwa' )
+		);
+		$insecure_content_more_details = sprintf(
+			'<a href="%s">%s</a>',
+			'https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content/How_to_fix_website_with_mixed_content',
+			esc_html__( 'More details', 'pwa' )
+		);
+
+		/**
+		 * Todo: change this description based on what insecure content is found.
+		 *
+		 * Have a separate description for when there's only passive insecure content.
+		 * And one for when there's active and passive insecure content.
+		 *
+		 * @see https://github.com/xwp/pwa-wp/issues/17#issuecomment-406188455
+		 */
+		/* Translators: %s: a link for more details */
+		$insecure_content_description = esc_html__( 'Your home page doesn’t contain insecure URLs. However, there may be URLs on other pages that could be blocked. %s', 'pwa' );
```

Comment:
Using a UTF-8 `’` in the string can have unintended effects on non-UTF-8 installs. It should be an HTML entity instead. For instance: https://github.com/WordPress/wordpress-develop/blob/927cd42632207fb78cf463a3a3f6df4bcd58e7b9/src/wp-includes/class-wp-customize-nav-menus.php#L1028

Reply:
Thanks, [fbd258](https://github.com/xwp/pwa-wp/pull/36/commits/fbd25818243cbd6f763b7b7406c318c9ba64ba20) uses the character `&#8217;`.